test/Helper/HttpProxyTest.php

```diff
@@ -0,0 +1,348 @@
+    }
+
+    /**
+     * @test
+     */
+    public function it_proxies_a_request()
+    {
+        $this->storage->save(
+            new Request(
+                'GET',
+                'http://www.example.com/test.mp3',
+                [
+                    'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+                    'Referer' => 'http://www.example.com/',
+                    'X-Forwarded-For' => '127.0.0.1',
+                    'X-Forwarded-Host' => 'localhost',
+                    'X-Forwarded-Port' => '80',
+                    'X-Forwarded-Proto' => 'http',
+                ]
+            ),
+            new Response(
+                200,
+                ['Content-Type' => 'audio/mp3'],
+                fopen($mp3 = __DIR__.'/../../assets/tests/blank.mp3', 'r')
+            )
+        );
+
+        $request = HttpFoundationRequest::create('GET', 'http://www.example.com/');
+
+        $response = $this->httpProxy->send($request, 'http://www.example.com/test.mp3');
+
+        $this->assertInstanceOf(StreamedResponse::class, $response);
+        $this->assertSame(200, $response->getStatusCode());
+        $this->assertArraySubset([
+            'content-type' => ['audio/mp3'],
+        ], $response->headers->all());
+        $this->assertSame(file_get_contents($mp3), $this->captureContent($response));
+    }
+
+    /**
+     * @test
+     */
+    public function it_sets_x_forwarded_for_when_not_through_a_trusted_proxy()
+    {
+        $this->storage->save(
+            new Request(
+                'GET',
+                'http://www.example.com/test.txt',
+                [
+                    'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+                    'X-Forwarded-For' => '127.0.0.1',
+                    'X-Forwarded-Host' => 'localhost',
+                    'X-Forwarded-Port' => '80',
+                    'X-Forwarded-Proto' => 'http',
+                ]
+            ),
+            new Response(
+                200,
+                ['Content-Type' => 'text/plain'],
+                'test'
+            )
+        );
+
+        $request = HttpFoundationRequest::create('GET', 'http://www.example.com/', [], [], [], ['HTTP_X_FORWARDED_FOR' => '54.230.78.56, 34.197.12.171']);
+
+        $response = $this->httpProxy->send($request, 'http://www.example.com/test.txt');
+
+        $this->assertInstanceOf(StreamedResponse::class, $response);
+        $this->assertSame(200, $response->getStatusCode());
+        $this->assertArraySubset([
+            'content-type' => ['text/plain'],
+        ], $response->headers->all());
+        $this->assertSame('test', $this->captureContent($response));
+    }
+
+    /**
+     * @test
+     */
+    public function it_adds_x_forwarded_for_when_through_a_trusted_proxy()
+    {
+        HttpFoundationRequest::setTrustedProxies(['127.0.0.1'], HttpFoundationRequest::HEADER_X_FORWARDED_ALL);
```

Comment:
:policeman: global state that is not reset during teardown of this test

Reply:
(unless it's in a parent class)

Reply:
Ah yes (was copied from elsewhere, so good to have fresh eyes on it).